[Main < T0878-DX] Add aditional content (#3)
dtomicevic/neo4jperf-guide: docs/indexing.md

slug: /neo4j-slow-query-performance-issue
---

> Ahh, indexing! One of the most common pitfalls in database performance in general, Neo4j included, is the usage of indexes. At first, it may seem what I need an index for in the Graph database, but graphs can be stored as some other data structure in the background, this means you can index and optimize for storage data structure. The wrong usage of the index can make a database unusable, which can be quite painful. Let's see a few common issues with Neo4j database performance and indexes. 

Reviewer: ```suggestion
Ahh, indexing! One of the most common pitfalls in database performance in general, Neo4j included, is the usage of indexes. At first, it may seem that indices are unnecessary in graph databases, but graphs can be stored as some other data structure in the background, which means you can index and optimize for the storage data structure. The wrong usage of the index can make a database unusable, which can be quite painful. Let's see a few common issues with Neo4j database performance and indexes. 
```

Reviewer: In line 13, this sentence doesn't make sense: 
As it turns out, this Neo4j user didn't even set up the trigger, so the search is not very slow because it is brute forcing across the database. 

Line 16:
Turns out the user.... **NOT Turn's out**

Reviewer: Line 39:
In your case, make sure that when you use the node that has an index, also use the property your index is on. 

Maybe: In your case, make sure that when you use the node that has an index, you also use the property the index is on.
